docs/src/advanced-concepts/preparing-generic-targets.rst

```diff
@@ -0,0 +1,110 @@
+like this:
+
+.. code-block:: yaml
+
+    targets:
+      mtt::my_target:
+        quantity: ""
+        read_from: dataset.xyz
+        key: energy
+        unit: ""
+        per_atom: True
+        type:
+          cartiesian:
+            rank: 1
+        num_properties: 10
+
+The crucial fields here are:
+
+- ``per_atom``: This field should be set to ``True`` if the target is a per-atom
+    property. Otherwise, it should be set to ``False``.
+- ``type``: This field specifies the type of the target. In this case, the target is
+    a Cartesian vector. The ``rank`` field specifies the rank of the target. For
+    Cartesian vectors, the rank is 1. Other possibilities are ``scalar`` and
+    ``spherical`` (for a spherical tensor).
+- ``num_properties``: This field specifies the number of different properties in the
+    target that need to be learned. In this case, there are 10 properties.
+
+A few more words should be spent on ``spherical`` targets. These should be made of a
+certain number of irreducible spherical tensors. For example, if you are learning a
+property that can be decomposed into two spherical tensors with L=0 and L=2, the target
+section should would look like this:
+
+.. code-block:: yaml
+
+    targets:
+      mtt::my_target:
+        quantity: ""
+        read_from: dataset.xyz
+        key: energy
+        unit: ""
+        per_atom: True
+        type:
+          spherical:
+            irreps:
+                - o3_lambda: 0
+                  o3_sigma: 1
+                - o3_lambda: 2
+                  o3_sigma: 1
+        num_properties: 10
+
+where ``o3_lambda`` specifies the L value of the spherical tensor and ``o3_sigma`` its
+parity with respect to inversion (1 for proper tensors, -1 for pseudo-tensors).
+
+Preparing your targets -- ASE
+-----------------------------
+
+If you are using the ASE readers to read your targets, you will have to save them
+either in the ``.info`` (if the target is per structure, i.e. not per atom) or in the
+``.arrays`` (if the target is per atom) attributes of the ASE atoms object. Then you can
+dump the atoms object to a file using ``ase.io.write``.
+
+The ASE reader will automatically try to reshape the target data to the format expected
+given the target section in the input file. In case your target data is invalid, an
+error will be raised.
+
+Reading targets with more than one spherical tensor is not supported by the ASE reader.
+In that case, you should use the metatensor reader.
+
+Preparing your targets -- metatensor
+------------------------------------
+
+If you are using the metatensor readers to read your targets, you will have to save them
+as a list of ``metatensor.torch.TensorMap`` objects with ``torch.save`` into a file with
+the ``.mts`` extension.
+
+The metatensor reader will verify that the target data in the input files corresponds to
+the metadata in the provided ``TensorMap`` objects. In case of a mismatch, errors will
+be raised.
```

Comment:
This is possible but very complicated. `metatrain` must handle cases where there is more than one property (then do you read them? With what names? Do you prepend `mtt::`? Etc. Guessing the number of properties and symmetries isn't easy either because `per_atom` properties in ASE must be flattened)

Reply:
Sorry, bad example. I was thinking of the case where we read from metatensor data, which should already contain all this information, i.e. allowing 

```
mtt::my_target:
        read_from: dataset.npz
        quantity: ""
        unit: ""
```

Reply:
I don't know if inferring is a good feature to have yet. I would first make sure it works and then we can add it in the future if there is a need for it. This type of dataset is meant for experienced users (for the moment), who can be deliberate about their usage

Reply:
There is no inferring here, the data is already there. It is more about not duplicating information. In the current version, either you copy-paste what's already in your files, or you get an error.